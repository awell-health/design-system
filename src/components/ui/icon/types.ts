import {
  RiArrowUpSLine,
  RiThumbUpFill,
  RiThumbUpLine,
  RiThumbDownFill,
  RiThumbDownLine,
  RiLineChartLine,
  RiStickyNote2Fill,
  RiDeleteBinLine,
  RiDraggable,
  RiPuzzleFill,
  RiPriceTag3Fill,
  RiEdit2Fill,
  RiFileCopyFill,
  RiBookmarkLine,
  RiDeleteBin6Fill,
  RiMore2Line,
  RiCloseLargeLine,
  RiLayout3Line,
  RiTerminalBoxFill,
  RiTerminalWindowFill,
  RiCalculatorFill,
  RiCheckboxMultipleFill,
  RiHealthBookFill,
  RiSurveyFill,
  RiMessage3Fill,
  RiQuestionFill,
  RiAlertFill,
  RiArrowDropRightLine,
  RiArrowLeftRightFill,
  RiArrowDownSLine,
  RiArrowLeftRightLine,
  RiAddFill,
  RiErrorWarningLine,
  RiAccountPinBoxFill,
  RiDeleteBinFill,
  RiCloseLine,
  RiCloseFill,
  RiCursorFill,
  RiNodeTree,
  RiSearchLine,
  RiContractRightLine,
  RiErrorWarningFill,
  RiInformation2Fill,
  RiCheckboxFill,
  RiUploadCloudFill,
  RiPlayFill,
  RiStopCircleFill,
  RiArrowLeftSLine,
  RiArrowRightSLine,
  RiSparklingFill,
  RiExternalLinkLine,
  RiCheckFill,
  RiCheckLine,
  RiArrowLeftLine,
  RiArrowRightLine,
  RiArrowDownLine,
  RiArrowUpLine,
  RiQuestionLine,
  RiInformationLine,
  RiAlertLine,
  RiSubtractFill,
  RiUserFill,
  RiUserLine,
  RiHomeLine,
  RiEditCircleLine,
  RiPulseFill,
  RiArrowLeftDoubleFill,
  RiLifebuoyLine,
  RiCodeFill,
  RiSettings4Line,
  RiLogoutBoxRLine,
  RiText,
  RiHashtag,
  RiCoinLine,
  RiCalendarLine,
  RiBracesLine,
  RiPhoneLine,
  RiListCheck,
  RiListOrdered2,
  RiDatabase2Line,
  RiListCheck3,
  RiStethoscopeLine,
  RiTestTubeLine,
  RiTestTubeFill,
  RiHeartPulseLine,
  RiDossierLine,
} from '@remixicon/react';

export const ICONS = {
  RiArrowUpSLine,
  RiThumbUpFill,
  RiThumbUpLine,
  RiThumbDownFill,
  RiThumbDownLine,
  RiLineChartLine,
  RiStickyNote2Fill,
  RiDeleteBinLine,
  RiDraggable,
  RiPuzzleFill,
  RiPriceTag3Fill,
  RiEdit2Fill,
  RiFileCopyFill,
  RiBookmarkLine,
  RiDeleteBin6Fill,
  RiMore2Line,
  RiCloseLargeLine,
  RiLayout3Line,
  RiTerminalBoxFill,
  RiTerminalWindowFill,
  RiCalculatorFill,
  RiCheckboxMultipleFill,
  RiHealthBookFill,
  RiSurveyFill,
  RiMessage3Fill,
  RiQuestionFill,
  RiAlertFill,
  RiArrowDropRightLine,
  RiArrowLeftRightFill,
  RiArrowDownSLine,
  RiArrowLeftRightLine,
  RiAddFill,
  RiErrorWarningLine,
  RiAccountPinBoxFill,
  RiDeleteBinFill,
  RiCloseLine,
  RiCloseFill,
  RiCursorFill,
  RiNodeTree,
  RiSearchLine,
  RiContractRightLine,
  RiErrorWarningFill,
  RiInformation2Fill,
  RiCheckboxFill,
  RiUploadCloudFill,
  RiPlayFill,
  RiStopCircleFill,
  RiArrowLeftSLine,
  RiArrowRightSLine,
  RiSparklingFill,
  RiExternalLinkLine,
  RiCheckFill,
  RiCheckLine,
  RiArrowLeftLine,
  RiArrowRightLine,
  RiArrowDownLine,
  RiArrowUpLine,
  RiQuestionLine,
  RiInformationLine,
  RiAlertLine,
  RiSubtractFill,
  RiUserFill,
  RiUserLine,
  RiHomeLine,
  RiEditCircleLine,
  RiPulseFill,
  RiArrowLeftDoubleFill,
  RiLifebuoyLine,
  RiCodeFill,
  RiSettings4Line,
  RiLogoutBoxRLine,
  RiText,
  RiHashtag,
  RiCoinLine,
  RiCalendarLine,
  RiBracesLine,
  RiPhoneLine,
  RiListCheck,
  RiListCheck3,
  RiListOrdered2,
  RiDatabase2Line,
  RiStethoscopeLine,
  RiTestTubeLine,
  RiTestTubeFill,
  RiHeartPulseLine,
  RiDossierLine,
};

export enum IconSize {
  xs = 16,
  s = 20,
  m = 24,
  l = 28,
  xl = 32
}

export type IconType = keyof typeof ICONS;
