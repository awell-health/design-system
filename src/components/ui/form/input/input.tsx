import * as React from 'react';

import { cn } from '@/lib/utils';

interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {
  label?: string;
  helpText?: string;
  suffixIcon?: JSX.Element;
  prefixIcon?: JSX.Element;
  hasError?: boolean;
  onChange?: React.ChangeEventHandler<HTMLInputElement>;
  className?: string;
  type?: string;
  labelClassName?: string;
}
const Input = React.forwardRef<HTMLInputElement, InputProps>(
  (
    {
      className,
      label,
      helpText,
      suffixIcon,
      prefixIcon,
      hasError,
      onChange,
      type = 'text',
      labelClassName,
      ...props
    },
    ref
  ) => {
    const renderIcon = (icon: JSX.Element, position: 'left' | 'right') => {
      return (
        <span
          className={cn(
            'absolute h-[40px] flex items-center',
            position === 'left' && 'left-2.5',
            position === 'right' && 'right-2.5'
          )}
        >
          {React.cloneElement(icon, {
            size: '16',
            className: hasError ? 'fill-red-600' : 'fill-slate-500'
          })}
        </span>
      );
    };

    return (
      <div className=''>
        {label && (
          <div className={cn('label', labelClassName)}>
            <span className=' label-text text-slate-600 text-sm font-medium'>{label}</span>
          </div>
        )}
        <label className='form-control w-full flex relative'>
          {prefixIcon && renderIcon(prefixIcon, 'left')}
          <input
            type={type}
            className={cn(
              type !== 'file' && 'px-3.5 py-2.5',
              'h-[38px] min-h-[38px] rounded-lg gap-2 w-full border-box',
              'text-sm font-normal',
              'shadow border border-slate-300 text-black placeholder:text-slate-400',
              'focus:outline-0 focus:border focus:border-blue-500',
              'disabled:bg-slate-50 disabled:text-slate-300',
              prefixIcon && 'pl-8',
              suffixIcon && 'pr-8',
              hasError && 'border-red-500 focus:border-red-500',
              type === 'file' && 'file-input file-input-bordered file-input-primary',
              className
            )}
            onChange={onChange}
            ref={ref}
            {...props}
          />
          {suffixIcon && renderIcon(suffixIcon, 'right')}
        </label>
        {helpText && (
          <div className='label'>
            <span
              className={cn(
                'label-text-alt text-slate-500 text-xs font-normal',
                hasError && 'text-red-600'
              )}
            >
              {helpText}
            </span>
          </div>
        )}
      </div>
    );
  }
);
Input.displayName = 'Input';

export { Input };
