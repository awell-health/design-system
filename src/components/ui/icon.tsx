import * as React from "react";
import * as RemixIcon from "@remixicon/react";

export enum IconSize {
  xs = 16,
  s = 20,
  m = 24,
  l = 28,
  xl = 32,
}

export type IconType = keyof typeof RemixIcon;

interface Props {
  icon: IconType;
  size?: IconSize;
  color?: string;
  className?: string;
}

const Icon = (props: Props): React.JSX.Element => {
  const { icon, color = "default", className = "inline", size = IconSize.m } = props

  const IconComponent = RemixIcon[icon]

  return <IconComponent size={size} color={color} className={className} />;
};

export { Icon };
