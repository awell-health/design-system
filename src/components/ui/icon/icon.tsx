import * as React from 'react';
import { ICONS, IconSize, IconType } from './types';

interface Props {
  icon: IconType;
  size?: IconSize;
  color?: string;
  className?: string;
}

const Icon = (props: Props): React.JSX.Element => {
  const { icon, color = 'default', className = 'inline', size = IconSize.m } = props;

  const IconComponent = ICONS[icon];

  return <IconComponent size={size} color={color} className={className} />;
};

export { Icon };
