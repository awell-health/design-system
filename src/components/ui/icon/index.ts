import { Icon } from './icon';
import { IconType, IconSize } from './types';

export { Icon, type IconType, IconSize };
