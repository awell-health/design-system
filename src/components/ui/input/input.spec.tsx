import { render } from '@testing-library/react';
import { expect, it, describe } from 'vitest';
import { Input } from './input';
import { Icon } from '../icon';

describe('', () => {
  it('match snapshot', () => {
    const props = {
      label: 'Label',
      helpText: 'Help text',
      prefixIcon: <Icon icon='RiAddFill' />,
      suffixIcon: <Icon icon='RiArrowLeftLine' />
    };
    const result = render(<Input {...props} />);

    expect(result).toMatchSnapshot();
  });

  it('displays error', () => {
    const value = 'test input value';
    const { getByDisplayValue } = render(<Input hasError={true} value={value} />);

    expect(getByDisplayValue(value)).toHaveClass('border-red-500');
  });
});
