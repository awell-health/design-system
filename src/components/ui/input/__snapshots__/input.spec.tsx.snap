// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=""
      >
        <div
          class="label"
        >
          <span
            class=" label-text text-slate-600 text-sm font-medium"
          >
            Label
          </span>
        </div>
        <label
          class="form-control w-full flex relative"
        >
          <span
            class="absolute h-[40px] flex items-center left-2.5"
          >
            <svg
              class="remixicon fill-slate-500"
              fill="default"
              height="16"
              viewBox="0 0 24 24"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11 11V5H13V11H19V13H13V19H11V13H5V11H11Z"
              />
            </svg>
          </span>
          <input
            class="px-3.5 py-2.5 h-[38px] min-h-[38px] rounded-lg gap-2 w-full text-sm font-normal shadow border border-slate-300 text-black placeholder:text-slate-400 focus:outline-0 focus:border focus:border-blue-500 disabled:bg-slate-50 disabled:text-slate-300 pl-8 pr-8"
            type="text"
          />
          <span
            class="absolute h-[40px] flex items-center right-2.5"
          >
            <svg
              class="remixicon fill-slate-500"
              fill="default"
              height="16"
              viewBox="0 0 24 24"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.82843 10.9999H20V12.9999H7.82843L13.1924 18.3638L11.7782 19.778L4 11.9999L11.7782 4.22168L13.1924 5.63589L7.82843 10.9999Z"
              />
            </svg>
          </span>
        </label>
        <div
          class="label"
        >
          <span
            class="label-text-alt text-slate-500 text-xs font-normal"
          >
            Help text
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=""
    >
      <div
        class="label"
      >
        <span
          class=" label-text text-slate-600 text-sm font-medium"
        >
          Label
        </span>
      </div>
      <label
        class="form-control w-full flex relative"
      >
        <span
          class="absolute h-[40px] flex items-center left-2.5"
        >
          <svg
            class="remixicon fill-slate-500"
            fill="default"
            height="16"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 11V5H13V11H19V13H13V19H11V13H5V11H11Z"
            />
          </svg>
        </span>
        <input
          class="px-3.5 py-2.5 h-[38px] min-h-[38px] rounded-lg gap-2 w-full text-sm font-normal shadow border border-slate-300 text-black placeholder:text-slate-400 focus:outline-0 focus:border focus:border-blue-500 disabled:bg-slate-50 disabled:text-slate-300 pl-8 pr-8"
          type="text"
        />
        <span
          class="absolute h-[40px] flex items-center right-2.5"
        >
          <svg
            class="remixicon fill-slate-500"
            fill="default"
            height="16"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.82843 10.9999H20V12.9999H7.82843L13.1924 18.3638L11.7782 19.778L4 11.9999L11.7782 4.22168L13.1924 5.63589L7.82843 10.9999Z"
            />
          </svg>
        </span>
      </label>
      <div
        class="label"
      >
        <span
          class="label-text-alt text-slate-500 text-xs font-normal"
        >
          Help text
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
